### DISCUSSION TITLE
how to setup the environment for debugging the metals

### DISCUSSION THREAD

**Discussion by tsxjsx** [scalameta/metals#6203] [Q&A]
i want to debug the metals server code what was the procedure for debugging metals

**Comment by tgodzik** [comment]
You would need to add a configuration in Metals' `.vscode/launch.json`:
```

    {
      "type": "scala",
      "request": "attach",
      "name": "Debug Metals",
      "buildTarget": "metals",
      "hostName": "localhost",
      "port": 5005
    }
```
and to the actual debugged workspace add to the workspace's settings `metals.serverProperties` `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005`

**Reply by tgodzik** [reply]
Ach, forgot about it, but you also need the quite flag. I recommend using the docs for more information https://scalameta.org/metals/docs/contributors/getting-started#jvm-debugging There is a search option.

And only if that is missing raise the discussion here so that we can update the docs.

**Reply by tsxjsx** [reply]
srry for late reply tgodzik,
I've figured out how to set up the environment for debugging the metals, and I've also identified an issue with metals, which I'll be posting in the issue section shortly.
again thanks tgodzik
